interface/include/nakama-cpp/NUtils.h

```diff
@@ -16,18 +16,56 @@
 
 #pragma once
 
-#include <nakama-cpp/NTypes.h>
 #include <nakama-cpp/NExport.h>
+#include <nakama-cpp/NTypes.h>
 
 NAKAMA_NAMESPACE_BEGIN
 
-    /**
-     * Get current UNIX time in milliseconds.
-     * 
-     * Returns number of milliseconds that have elapsed since 00:00:00 Thursday, 1 January 1970.
-     * 
-     * @return UNIX time in milliseconds.
-     */
-    NAKAMA_API NTimestamp getUnixTimestampMs();
+/**
+ * Get current UNIX time in milliseconds.
+ *
+ * Returns number of milliseconds that have elapsed since 00:00:00 Thursday, 1 January 1970.
+ *
+ * @return UNIX time in milliseconds.
+ */
+NAKAMA_API NTimestamp getUnixTimestampMs();
+
+struct disarmed_t {};
+inline constexpr disarmed_t disarmed{};
+
+// like unique_ptr but for non-pointer types
+template <typename T, int (*d)(T)> class Resource {
```

Comment:
this header is part of API and get shipped with binary builds. Resource is not used anywhere in API and shouldn't be here, I'll move it to implementation